plugins/destination/mssql/client/migrate.go

```diff
@@ -59,26 +59,36 @@ func (c *Client) schemaTables(ctx context.Context, tables schema.Tables) (schema
 	return schemaTables, nil
 }
 
-func (c *Client) normalizedTables(tables schema.Tables) schema.Tables {
-	var normalized schema.Tables
-	for _, table := range tables.FlattenTables() {
-		table := table
-		for i := range table.Columns {
-			// Since multiple schema types can map to the same MSSQL type we need to normalize them to avoid false positives when detecting schema changes
-			// This should never return an error
-			typ, _ := queries.SchemaType(table.Name, table.Columns[i].Name, queries.SQLType(table.Columns[i].Type))
-			table.Columns[i].Type = typ
-		}
-		// If there are no PKs, we use CqID as PK
-		pks := table.PrimaryKeys()
-		if !c.pkEnabled() || len(pks) == 0 {
-			table.Columns.Get(schema.CqIDColumn.Name).CreationOptions.PrimaryKey = true
-		}
-
-		normalized = append(normalized, table)
+func (c *Client) normalizeTable(table *schema.Table) {
+	for i := range table.Columns {
+		// Since multiple schema types can map to the same MSSQL type we need to normalize them to avoid false positives when detecting schema changes
+		// This should never return an error
+		typ, _ := queries.SchemaType(table.Name, table.Columns[i].Name, queries.SQLType(table.Columns[i].Type))
+		table.Columns[i].Type = typ
+
+		if !c.pkEnabled() {
+			table.Columns[i].CreationOptions.PrimaryKey = false
+		}
+		if table.Columns[i].CreationOptions.PrimaryKey {
+			table.Columns[i].CreationOptions.NotNull = true
+		}
+
+		// this is for backward compatibility as I believe this is as of right now defined on the source
+		if c.pkEnabled() && len(table.PrimaryKeys()) == 0 {
+			cqIdColumn := table.Columns.Get(schema.CqIDColumn.Name)
+			if cqIdColumn != nil {
+				cqIdColumn.CreationOptions.PrimaryKey = true
+			}
+		}
 	}
+}
 
-	return normalized
+func (c *Client) normalizeTables(tables schema.Tables) schema.Tables {
```

Comment:
```suggestion
func (c *Client) normalizedTables(tables schema.Tables) schema.Tables {
```

so that the call is unchanged

Reply:
I copy-pasted the same name we have in postgresql